### DISCUSSION TITLE
Non-included script indexing/use

### DISCUSSION THREAD

**Discussion by tinygiant98** [PhilippeChab/nwscript-ee-language-server#32] [Ideas]
I understand fully the ease of indexing all of the available scripts that aren't in the ignore path, and I'm not even sure this would be desirable behavior, but I'd like to get your take on it.

1) Is it possible/desirable to have some kind of warning if user is adding functions from an #include file that isn't in the include chain for this specific script?

For example, if I have added a function from my "core_inc.nss", and this file is indexed because it exists in my project, but not #included in the script I'm currently working on, is it possible to add a warning (yellow squiggly) to let the user know he's adding a function from a non-included file, but still provide all the normal lsp functionality for said function?

2) Alternately, would it be desirable to remove all indexing/access to any script that's not in the #include chain for the script that's currently being edited?  That is, if I'm trying to pull in a function from an indexed script, but that script isn't #included in the file I'm currently editing, should lsp functionality for that file be blocked as kind of a hint that this script doesn't have access to those functions?

3) Or, just let nwnsc do its thing and let the user be smart enough to figure it out.

**Comment by PhilippeChab** [comment]
Hey! 

Thanks for having so much enthusiasm for the project!

> Is it possible/desirable to have some kind of warning if user is adding functions from an #include file that isn't in the include chain for this specific script?

This would be easy to do, however I'm not sure it is a good idea as a) it will hurt performances of the diagnostics publishing process - it can already take half a second to a second to compile a big file on an older machine - and b) the compiler already does that for us.

> That is, if I'm trying to pull in a function from an indexed script, but that script isn't #included in the file I'm currently editing, should lsp functionality for that file be blocked as kind of a hint that this script doesn't have access to those functions?

That is the current behaviour. :) Hover, goto and completion is only available for the local definitions, include definitions and `nwscript.nss` definitions.

**Reply by tinygiant98** [reply]
Okay, thanks for the reply.  I thought I was running into issues that allowed scripts well outside the include chain to be available for autocomplete, but I'll go back and run some tests to confirm. Thanks again.
